Tests/Functional/Controller/CalendarControllerTest.php

```diff
@@ -55,6 +55,119 @@ public function testCalendarsCreateAction()
         $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
         $this->assertCount(1, $crawler->filter('html:contains("Le calendrier a été créé")'));
     }
+    
+    /**
+     *  @group editCalendar
+     */
+    public function testCalendarsEditAction(){
+        $route = $this->generateRoute('canal_tp_mtt_calendars_list');
+
+        $crawler = $this->doRequestRoute($route);
+
+        $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
+
+        // Assert that there is a calendar in the calendars list
+        $calendarsFirstRow = $crawler->filter('#main-container table tbody tr')->first();
+        $this->assertCount(1, $calendarsFirstRow);
+
+        // Assert that edit button exists for a calendar
+        $calendarsFirstRowEditButton = $calendarsFirstRow->filter('td a[href*="edit"]');
+        $this->assertCount(1, $calendarsFirstRowEditButton);
+
+        // Assert that edit button redirect to calendar edit page
+        $calendarsFirstRowEditLink = $calendarsFirstRowEditButton->link();
+        $crawler = $this->client->click($calendarsFirstRowEditLink);
+        $expectedUri = $calendarsFirstRowEditLink->getUri();
+        $this->assertEquals($calendarsFirstRowEditLink->getUri(), $this->client->getRequest()->getUri());
```

Comment:
not sure of this test

you click on a link
and you test uri = link

it seems normal

you should test link = '/mtt/edit/6' (for example) before clicking